packages/rum-core/src/domain/tracing/tracer.ts

```diff
@@ -42,6 +42,10 @@ export function startTracer(configuration: Configuration): Tracer {
           Object.keys(context.init.headers).forEach((key) => {
             headers.push([key, (context.init!.headers as Record<string, string>)[key]])
           })
+        } else if (context.input instanceof Request) {
+          context.input.headers.forEach((value, key) => {
+            headers.push([key, value])
+          })
         }
         context.init.headers = headers.concat(objectEntries(tracingHeaders) as string[][])
```

Comment:
It seems a bit weird to me to not update input headers but to provide an init headers instead.
I am wondering if it could not be better to provide an updated input only:
```ts
context.input = new Request(request, {headers: headers.concat(objectEntries(tracingHeaders) as string[][])})
```
no strong argument for one or the other though.
wdyt?

Reply:
Mmh it seems to be equivalent, since calling [`fetch(input, init)` calls `new Request(input, init)` internally](https://fetch.spec.whatwg.org/#fetch-method). After a few tests, I think that it should be fine. I only spoted an issue with [streaming requests](https://web.dev/fetch-upload-streaming/) but it is likely to be a Chrome bug in the ongoing implementation:

```ts
const request = new Request('/post', {
  method: 'POST',
  body: new ReadableStream(),
  mode: 'cors'
});
// works:
await fetch(request, { mode: 'cors' })
// fails with: TypeError: Failed to construct 'Request': If request is made from ReadableStream, mode should be"same-origin" or "cors"
const requestCopy = new Request(request)
await fetch(requestCopy, { mode: 'cors' })
```

I think it's fine nonetheless :)